components/cam/src/physics/cam/bfb_math.inc

```diff
@@ -31,6 +32,7 @@
 #  define bfb_log(val) log(val)
 #  define bfb_log10(val) log10(val)
 #  define bfb_exp(val) exp(val)
+#  define bfb_expm1(val) (1.0 - exp(val))
```

Comment:
@ambrad and/or @jgfouca : expm1 doesn't seem to exist in fortran (at least on intel), so I'm defining it in the line above to be 1-exp(x) for E3SMv0 builds. This shouldn't affect scream tests even if this kludge implies a different algorithm because scream tests will always use the C++ version of expm1. My kludge will be less accurate in our SCREAMv0 runs when x is small, but I think that's ok because x=dt/tau_r in our case, which should never get too small for the applications we're currently using it for. Questions: 1). Does this hack seem ok to you? 2). I'm worried that the 1.0 here should be 1.0_rtype but that rtype isn't available here... which could cause casting to single precision and resulting loss of accuracy. Am I correct to worry about this? Thanks.

Reply:
Related point/reminder to self: going the expm1 route requires defining expm1 on packs in ekat. I'm waiting for confirmation that expm1 is good before issuing the 1-line ekat PR needed to get expm1 working in this PR.

Reply:
@PeterCaldwell but you should make it `(exp(val) - 1)`, i.e., opposite sign to what you have. Just use `1`, not `1.0`. `1` is exactly representable in FP, so no concerns here.

Reply:
Re: expm1 on packs. Yes, that'll need to be impl-ed for packs.

Reply:
@ambrad - doh. Thanks for the 1-exp(x) should be exp(x) -1 correction. I'm surprised you suggest using `1` instead of `1.0`: since this is fortran code, I thought type got cast to the crudest of the 2 inputs to '+' in fortran... You're probably right, I'm just sanity checking...

Reply:
the point is that, unlike, say, 1/3, 1 is exactly representable in binary format, regardless of precision, so it doesn't matter.